python/services/monitorservice/dmod/test/test_monitor_service.py

```diff
@@ -0,0 +1,981 @@
+        service.run_monitor_check()
+        change = service._dequeue_monitored_change(conn_id)
+        while change is not None:
+            change = service._dequeue_monitored_change(conn_id)
+
+        # Now have there be another change to "monitor"
+        job = self._jobs[1]
+        original_status = job.status
+        get_next_one = False
+        for status in JobStatus:
+            if get_next_one:
+                job.status = status
+                break
+            elif status == original_status:
+                get_next_one = True
+
+        # Then monitor again
+        service.run_monitor_check()
+        new_changes = list()
+        change = service._dequeue_monitored_change(conn_id)
+        while change is not None:
+            new_changes.append(change)
+            change = service._dequeue_monitored_change(conn_id)
+
+        self.assertEqual(len(new_changes), 1)
+
+    # Test that after first changes conveyed, then something done, the right changed job is listed
+    def test_run_monitor_check_3_b(self):
+        service = self._services[1]
+        conn_id = service.register_connection(connection=self, jobs_of_interest=self._job_ids)
+        service.run_monitor_check()
+        change = service._dequeue_monitored_change(conn_id)
+        while change is not None:
+            change = service._dequeue_monitored_change(conn_id)
+
+        # Now have there be another change to "monitor"
+        job = self._jobs[1]
+        original_status = job.status
+        get_next_one = False
+        for status in JobStatus:
+            if get_next_one:
+                job.status = status
+                break
+            elif status == original_status:
+                get_next_one = True
+
+        # Then monitor again
+        service.run_monitor_check()
+        change = service._dequeue_monitored_change(conn_id)
+
+        self.assertEqual(change.job.job_id, job.job_id)
+
+    # Test that after first changes conveyed, then something done, the right original status is listed
+    def test_run_monitor_check_3_c(self):
+        service = self._services[1]
+        conn_id = service.register_connection(connection=self, jobs_of_interest=self._job_ids)
+        service.run_monitor_check()
+        change = service._dequeue_monitored_change(conn_id)
+        while change is not None:
+            change = service._dequeue_monitored_change(conn_id)
+
+        # Now have there be another change to "monitor"
+        job = self._jobs[1]
+        original_status = job.status
+        get_next_one = False
+        for status in JobStatus:
+            if get_next_one:
+                job.status = status
+                break
+            elif status == original_status:
+                get_next_one = True
+
+        # Then monitor again
+        service.run_monitor_check()
+        change = service._dequeue_monitored_change(conn_id)
+
+        self.assertEqual(change.original_status, original_status)
```

Comment:
Gonna have to read these tests over time, but I'll take your word for them for now.

Reply:
This basically tests that a _MonitoredChange_ instance created during `run_monitor_check()` gets the correct `original_status` value (i.e., the job's status prior to the observed change).  In particular, the test is making sure that `run_monitor_check()` acquires this properly and does not use either a static or stale value.